lib/main.js

```diff
@@ -131,6 +131,21 @@ export default {
         },
       }),
     );
+
+    this.subscriptions.add(
+      atom.workspace.observeTextEditors((editor) => {
+        editor.onDidSave(() => {
+          const validScope = editor.getCursors().some(cursor =>
+            cursor.getScopeDescriptor().getScopesArray().some(scope =>
+              grammarScopes.includes(scope)));
+          if (validScope && atom.config.get('linter-tslint.fixOnSave')) {
```

Comment:
Actually, we can avoid having another `textEditors` subscription by adding this logic [here](https://github.com/AtomLinter/linter-tslint/blob/master/lib/main.js#L94)